extensions/notebook/src/book/bookUndoRedoService.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface IBookUndoRedoElement {
+	readonly label: string;
+	undo(): Promise<void> | void;
+	redo(): Promise<void> | void;
+}
+
+export class BookUndoRedoService {
```

Comment:
I'd suggest making this service be more generic than just books. This isn't doing anything specific to books and it's very likely we'll be adding other non-book stuff to this at some point so might as well make it generic now